### PR TITLE
feat: add links in the homepage

### PR DESCRIPTION
**Pull request by CarmitKl**
Add 6 cards to the Homepage, which on-click navigate to pages in the website. Adds material-ui dependency to the project.
Defined to have 3 columns for medium+ screens, so It's preferable to have at least 3 cards (it can be any number).

Cards are defined by the file: src > config > home.json.

### CODE REVIEW

File: src/components/Homepage/HomepageFeatures.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import styles from './HomepageFeatures.module.css';
+import {Button, Grid} from '@mui/material';
+import {GlobalStyles} from '@mui/styled-engine';
+import HomepageCard from './HomepageCard';
+import homeLinksList from '../../config/home.json';
+import {useHistory} from '@docusaurus/router';
+
+export default function HomepageFeatures() {
+  const list = homeLinksList;
```
Comment (dan-ziv) [inline src/components/Homepage/HomepageFeatures.js:10]: redundant assignment

File: src/components/Homepage/HomepageFeatures.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          >
+            {list.map(({linkName, ...props}, idx) => {
+              const history = useHistory();
+
+              return (
+                <Grid key={idx} item xs={8} md={4}>
+                  <Button
+                    variant="contained"
+                    onClick={() => {
+                      console.log('linkName: ', linkName);
```
Comment (dan-ziv) [inline src/components/Homepage/HomepageFeatures.js:28]: consider moving event handlers outside and maybe even use useCallback

File: src/components/Homepage/HomepageFeatures.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                      history.push(linkName);
+                    }}
+                    onKeyDown={e => {
+                      if (e.key === 'Enter') {
+                        console.log('linkName: ', linkName);
+                        history.push(linkName);
+                      }
+                    }}
+                    sx={{
```
Comment (dan-ziv) [inline src/components/Homepage/HomepageFeatures.js:37]: consider moving style objects outside the JSX

File: src/components/Homepage/HomepageCard.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import {CardContent, CardMedia, Typography} from '@mui/material';
+import {GlobalStyles} from '@mui/styled-engine';
+
+const HomepageCard = ({title, text, img}) => {
+  console.log('got img: ', img);
+  const blue = '#29296E';
```
Comment (dan-ziv) [inline src/components/Homepage/HomepageCard.js:7]: export this color from css file?

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: src/components/HomepageFeatures/HomepageFeatures.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  width: '100%',
+  minHeight: '100%',
+  backgroundColor: '#fff',
+  [':hover']: {
+    boxShadow: '-1px 5px 19px 0px rgba(0,0,0,0.3)',
+    backgroundColor: '#fff'
+  }
+};
+
+const renderCards = () => {
```
Comment (dan-ziv) [inline src/components/HomepageFeatures/HomepageFeatures.js:19]: Move `renderCards` inside the `HomepageFeatures` component

File: src/components/HomepageFeatures/HomepageFeatures.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  const history = useHistory();
```
Comment (dan-ziv) [inline src/components/HomepageFeatures/HomepageFeatures.js:20]: Hooks should always be placed at the top of the component and not inside inner functions. Please move to `HomepageFeatures`

File: src/components/HomepageFeatures/HomepageFeatures.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  return list.length
+    ? list.map(({linkName, ...props}, idx) => (
+        <Grid key={idx} item xs={8} md={4}>
+          <Button
+            variant="contained"
+            onClick={() => handleOnClick(linkName)}
+            onKeyDown={e => handleOnKeyDown(e, linkName)}
+            sx={overrideLinkStyle}
+          >
+            <HomepageCard {...props}></HomepageCard>
```
Comment (dan-ziv) [inline src/components/HomepageFeatures/HomepageFeatures.js:36]:
```suggestion
            <HomepageCard {...props}/>
```

File: src/components/HomepageFeatures/HomepageFeatures.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+          </Button>
+        </Grid>
+      ))
+    : null;
+};
+
+const HomepageFeatures = () => {
```
Comment (dan-ziv) [inline src/components/HomepageFeatures/HomepageFeatures.js:43]:
```suggestion
export const HomepageFeatures = () => {
```

File: src/components/HomepageCard/HomepageCard.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import {CardContent, CardMedia, Typography} from '@mui/material';
+import {Blue} from '../../config/colors';
+
+const HomepageCard = ({title, text, img}) => {
```
Comment (dan-ziv) [inline src/components/HomepageCard/HomepageCard.js:5]:
```suggestion
export const HomepageCard = ({title, text, img}) => {
```

File: src/config/colors.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export const Blue = {
```
Comment (dan-ziv) [inline src/config/colors.js:1]: Why not in css file?

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: src/config/colors.js (hunk 10 above)
Comment (CarmitKl) [reply src/config/colors.js:1]: 'colors' should be reused between many files. Overriding mui-components is much easier using the 'sx' prop, than creating a ccs and importing it, and making sure in it to override relevant classes as seen on the browser (need to manually search the dom).

**Review by dan-ziv — CHANGES_REQUESTED** [review]

**Review by dan-ziv — APPROVED** [review]

**Pull request merged**